In [1]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
from astropy.table import Table
from astropy.coordinates import SkyCoord
import astropy.units as u

In [3]:
gal= Table.read('BOSSDAT.fits')
quasars= Table.read('qso.fits')

In [4]:
print gal, quasars

PLATE  MJD  FIBER         Z                RA            DEC        OBJC_TYPE
----- ----- ----- ------------------ ------------- ---------------- ---------
 3586 55181     1     0.465112239122 9.33191224076  -0.462954655955         3
 3586 55181     2     0.719726264477 9.33007790809  -0.624115895012         3
 3586 55181     3      0.58804833889 9.43210592532  -0.245181536013         3
 3586 55181     5     0.585018992424 9.36867467866  -0.339224012425         3
 3586 55181     6     0.532988071442 9.40915664424  -0.250057843342         3
 3586 55181     7     0.474424600601 9.36047036958  -0.212842368574         3
 3586 55181     8     0.489556133747 9.35193235597  -0.159622251105         3
 3586 55181     9     0.489935219288 9.45202747482  -0.105271414218         3
 3586 55181    10     0.464580029249 9.47772265588  -0.131829129093         3
 3586 55181    11     0.490401208401 9.47710888524 -0.0977252219324         3
  ...   ...   ...                ...           ...              

In [5]:
def dist(i,j):
    qa = SkyCoord(quasars['RA'][i], quasars['DEC'][i], "icrs", unit="deg")
    ga = SkyCoord(gal['RA'][j], gal['DEC'][j], "icrs", unit="deg") 
    return qa.separation(ga).to(u.arcsec).value

In [6]:
qatalog = SkyCoord(quasars['RA']*u.degree, quasars['DEC']*u.degree ) 
gatalog = SkyCoord(gal['RA']*u.degree, gal['DEC']*u.degree ) 

In [7]:
idx, d2d, d3d = qatalog.match_to_catalog_sky(gatalog)

In [8]:
closest = np.argsort(d2d)  
#argsort - sorts out
#d2d - The on-sky separation between the closest match for each element in this object
#idk - 

In [15]:
for k in range(0,2500,2):
    i=closest[k]
        #i=closest[k] - gets every other one so doesnt repeat
    j=idx[i]
        #idx - gets the pair of i
    if quasars['RA'][i] != gal['RA'][j] and quasars['DEC'][i] != gal['DEC'][j]:
        print i, j, dist(i,j)

69061 1216901 0.0123149981576
112238 1336615 2.02901631567
172579 1340827 2.43263115148
107193 450344 2.48716436514
63709 733457 2.70737935553
175237 1340923 2.75845839133
89480 1530356 2.81973848458
241940 1378028 3.00669097972
241730 626071 3.06344370029
137673 1359518 3.09323559564
165496 1488728 3.25354580049
208575 1415472 3.31611708217
134789 1355867 3.3665241954
16067 498548 3.53808713781
86516 929263 3.57465438692
170459 1340724 3.62145092271
17022 308301 3.63847162734
162773 1024073 3.68390999741
38124 1536259 3.7834580992
221695 1415072 3.80545570084
31873 264998 3.8515061316
222298 738103 3.96031485056
52728 1211921 4.02653615452
135921 1228708 4.09840245223
42710 20861 4.13759424569
35611 45925 4.30046425132
75399 486915 4.32063567824
187224 1317437 4.32269583624
124897 605163 4.394186655
23241 1309588 4.41379026105
149518 491641 4.47233662644
226003 1068038 4.521094353
204161 106028 4.53060544618
81115 783060 4.58125052919
159104 467738 4.61664506679
72652 487131 4.6395877

print quasars['PLATE'][69061], quasars['MJD'][69061], quasars['FIBER'][69061], gal['PLATE'][1216901], gal['MJD'][1216901], gal['FIBER'][1216901]

In [63]:
print quasars['PLATE'][34310], quasars['MJD'][34310], quasars['FIBER'][34310], gal['PLATE'][1518858], gal['MJD'][1518858], gal['FIBER'][1518858]

7236 56605 348 7236 56605 348
